### ISSUE TITLE
model_class_name = MODEL_TYPE_TO_MODEL_CLASS_NAME[model_type]

### ISSUE THREAD

**Issue by monkeycc** [ultralytics/ultralytics#18406]
### Search before asking

- [X] I have searched the Ultralytics YOLO [issues](https://github.com/ultralytics/ultralytics/issues) and found no similar bug report.

### Ultralytics YOLO Component

Predict

### Bug

```
detection_model = AutoDetectionModel.from_pretrained(
                      ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "D:\anaconda3\envs\py312\Lib\site-packages\sahi\auto_model.py", line 65, in from_pretrained
    model_class_name = MODEL_TYPE_TO_MODEL_CLASS_NAME[model_type]
```

### Environment

OS                  Windows-11-10.0.22631-SP0
Environment         Windows
Python              3.12.5
Install             pip
RAM                 63.67 GB
Disk                418.1/1024.0 GB
CPU                 12th Gen Intel Core(TM) i9-12900H
CPU count           20
GPU                 NVIDIA GeForce RTX 3080 Ti Laptop GPU, 16384MiB
GPU count           1
CUDA                11.8

numpy               ✅ 1.26.4>=1.23.0
numpy               ✅ 1.26.4<2.0.0; sys_platform == "darwin"
matplotlib          ✅ 3.9.1>=3.3.0
opencv-python       ✅ 4.10.0.84>=4.6.0
pillow              ✅ 11.0.0>=7.1.2
pyyaml              ✅ 6.0.1>=5.3.1
requests            ✅ 2.32.3>=2.23.0
scipy               ✅ 1.14.0>=1.4.1
torch               ✅ 2.3.1+cu118>=1.8.0
torch               ✅ 2.3.1+cu118!=2.4.0,>=1.8.0; sys_platform == "win32"
torchvision         ✅ 0.18.1+cu118>=0.9.0
tqdm                ✅ 4.66.4>=4.64.0
psutil              ✅ 6.0.0
py-cpuinfo          ✅ 9.0.0
pandas              ✅ 2.2.0>=1.1.4
seaborn             ✅ 0.13.2>=0.11.0
ultralytics-thop    ✅ 2.0.0>=2.0.0

### Minimal Reproducible Example

https://docs.ultralytics.com/zh/guides/sahi-tiled-inference/#instantiate-the-model

```python
from sahi import AutoDetectionModel

detection_model = AutoDetectionModel.from_pretrained(
    model_type="ultralytics",
    model_path="./yolov11.pt",
    confidence_threshold=0.3,
    device="cuda:0",  # or 'cuda:0'
)
```

### Are you willing to submit a PR?

- [ ] Yes I'd like to help by submitting a PR!

**Comment by UltralyticsAssistant** [comment]
👋 Hello @monkeycc, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com) where you can find a variety of [Python](https://docs.ultralytics.com/usage/python/) and [CLI](https://docs.ultralytics.com/usage/cli/) usage examples, which might address your concerns.

If this is a 🐛 **Bug Report**, could you please provide a more detailed [minimum reproducible example](https://docs.ultralytics.com/help/minimum_reproducible_example/) and any additional context you may find relevant? This will help us diagnose the issue more effectively. For instance:
- Verify the `model_path` and any prior steps involving model loading.
- Check whether the file `yolov11.pt` is correctly set up and accessible.

If you’re attempting a custom training ❓ **Question**, please share detailed training logs and outputs, as well as insights into the dataset structure and training configuration, while following our [Tips for Best Training Results](https://docs.ultralytics.com/guides/model-training-tips/).

Join the Ultralytics community to enhance your involvement with like-minded peers:
- 🎧 Real-time discussions on [Discord](https://discord.com/invite/ultralytics)  
- In-depth engagements on [Discourse](https://community.ultralytics.com)  
- Community threads on our [Subreddit](https://reddit.com/r/Ultralytics).

## **Upgrade**

Please ensure you're running the latest `ultralytics` package and all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml) in a [Python>=3.8](https://www.python.org/) environment with [PyTorch>=1.8](https://pytorch.org/get-started/locally/). Update your environment with:

```bash
pip install -U ultralytics
```

## **Environments**

YOLO can also be used in these verified environments with dependencies pre-installed:
- **Notebooks**: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** or **AWS**: Follow the [GCP](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/) and [AWS](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/) guides.
- **Docker**: Check the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) for deployment.<a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## **Status**

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml/badge.svg" alt="Ultralytics CI"></a>  
If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yaml?query=event%3Aschedule) tests are passing. These tests verify YOLO’s functionality on various platforms.

This is an automated response. An Ultralytics engineer will follow up with you shortly for further assistance! 😊

**Comment by Y-T-G** [comment]
This is not related to Ultralytics. It's an error in SAHI code. You should open an issue there.
